gcc/rust/ast/rust-expr.h

```diff
@@ -4620,6 +4621,23 @@ struct AnonConst
 {
   NodeId id;
   std::unique_ptr<Expr> value;
+  AnonConst () {}
+  AnonConst (const AnonConst &other)
+  {
+    id = other.id;
+    value = other.value == nullptr
+	      ? nullptr
+	      : std::unique_ptr<Expr> (other.value->clone_expr ());
```

Comment:
I think it'd be clearer to use a regular `if` here 
```suggestion
    if (other.value)
        value = other.value->clone_expr ();
```

and `value` should be set to `nullptr` automatically anyway